subsystem/metrics_test.go

```diff
@@ -0,0 +1,276 @@
+			}
+			err = wait.Poll(time.Millisecond, 10*time.Second, waitFunc)
+			Expect(err).NotTo(HaveOccurred())
+
+			metricsDeregisterCluster(ctx, clusterID)
+			assertValidationCounter(clusterID, models.HostValidationIDHasMinCPUCores, 1)
+			assertValidationCounter(clusterID, models.HostValidationIDHasMinMemory, 1)
+			//assertValidationCounter(clusterID, models.HostValidationIDHasMinValidDisks, 1)
+		})
+
+		It("machine-cidr-defined", func() {
+		})
+
+		It("role-defined", func() {
+		})
+
+		It("has-min-hw-capacity-for-role", func() {
+
+			host := &registerHost(clusterID).Host
+
+			var lowHwInfo models.Inventory
+			err := copier.CopyWithOption(&lowHwInfo, validHwInfo, copier.Option{DeepCopy: true})
+			Expect(err).NotTo(HaveOccurred())
+
+			lowHwInfo.CPU.Count = 1
+			lowHwInfo.Memory.PhysicalBytes = int64(4 * units.GiB)
+
+			generateHWPostStepReply(ctx, host, &lowHwInfo, "master-0")
+
+			waitFunc := func() (bool, error) {
+				cpuCond, _ := isHostValidationInFailedStatus(clusterID, *host.ID, models.HostValidationIDHasCPUCoresForRole)
+				memoryCond, _ := isHostValidationInFailedStatus(clusterID, *host.ID, models.HostValidationIDHasMemoryForRole)
+				return cpuCond && memoryCond, nil
+			}
+			err = wait.Poll(time.Millisecond, 10*time.Second, waitFunc)
+			Expect(err).NotTo(HaveOccurred())
+
+			metricsDeregisterCluster(ctx, clusterID)
+			assertValidationCounter(clusterID, models.HostValidationIDHasCPUCoresForRole, 1)
+			assertValidationCounter(clusterID, models.HostValidationIDHasMemoryForRole, 1)
+		})
+
+		It("hostname-unique", func() {
+
+			host1 := &registerHost(clusterID).Host
+			generateHWPostStepReply(ctx, host1, validHwInfo, "nonUniqName")
+
+			host2 := &registerHost(clusterID).Host
+			generateHWPostStepReply(ctx, host2, validHwInfo, "nonUniqName")
+
+			waitFunc := func() (bool, error) {
+				host1Cond, _ := isHostValidationInFailedStatus(clusterID, *host1.ID, models.HostValidationIDHostnameUnique)
+				host2Cond, _ := isHostValidationInFailedStatus(clusterID, *host2.ID, models.HostValidationIDHostnameUnique)
+				return host1Cond && host2Cond, nil
+			}
+			err := wait.Poll(time.Millisecond, 10*time.Second, waitFunc)
+			Expect(err).NotTo(HaveOccurred())
+
+			metricsDeregisterCluster(ctx, clusterID)
+			assertValidationCounter(clusterID, models.HostValidationIDHostnameUnique, 2)
+		})
+
+		It("hostname-valid", func() {
+
+			// 'localhost' is a forbidden host name
+			host := &registerHost(clusterID).Host
+			generateHWPostStepReply(ctx, host, validHwInfo, "localhost")
+
+			waitFunc := func() (bool, error) {
+				return isHostValidationInFailedStatus(clusterID, *host.ID, models.HostValidationIDHostnameValid)
+			}
+			err := wait.Poll(time.Millisecond, 10*time.Second, waitFunc)
+			Expect(err).NotTo(HaveOccurred())
+
+			metricsDeregisterCluster(ctx, clusterID)
+			assertValidationCounter(clusterID, models.HostValidationIDHostnameValid, 1)
+		})
+
+		It("belongs-to-machine-cidr", func() {
```

Comment:
what are those empty test cases?

Reply:
All the tests that needs to be filled here https://issues.redhat.com/browse/MGMT-3645

Reply:
then please add them to the ticket description instead of having dead code in this PR

Reply:
done.